services/motion/builtin/builtin_test.go

```diff
@@ -1731,25 +1279,897 @@ func TestMoveOnGlobe(t *testing.T) {
+
+		t.Run("ensure success of movement around obstacle", func(t *testing.T) {
+			kb, ms := createMoveOnMapEnvironment(ctx, t, "pointcloud/octagonspace.pcd", 40, nil)
+			defer ms.Close(ctx)
+
+			req := motion.MoveOnMapReq{
+				ComponentName: base.Named("test-base"),
+				Destination:   goalInSLAMFrame,
+				SlamName:      slam.Named("test_slam"),
+				Extra:         extra,
+			}
+
+			executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+
+			timeoutCtx, timeoutFn := context.WithTimeout(ctx, time.Second*5)
+			defer timeoutFn()
+			err = motion.PollHistoryUntilSuccessOrError(timeoutCtx, ms, time.Millisecond*5, motion.PlanHistoryReq{
+				ComponentName: req.ComponentName,
+				ExecutionID:   executionID,
+				LastPlanOnly:  true,
+			})
+			test.That(t, err, test.ShouldBeNil)
+
+			endPos, err := kb.CurrentPosition(ctx)
+			test.That(t, err, test.ShouldBeNil)
+
+			test.That(t, spatialmath.PoseAlmostCoincidentEps(endPos.Pose(), goalInBaseFrame, 15), test.ShouldBeTrue)
+		})
+		t.Run("check that straight line path executes", func(t *testing.T) {
+			kb, ms := createMoveOnMapEnvironment(ctx, t, "pointcloud/octagonspace.pcd", 40, nil)
+			defer ms.Close(ctx)
+			easyGoalInBaseFrame := spatialmath.NewPoseFromPoint(r3.Vector{X: 0.277 * 1000, Y: 0.593 * 1000})
+			easyGoalInSLAMFrame := spatialmath.PoseBetweenInverse(motion.SLAMOrientationAdjustment, easyGoalInBaseFrame)
+
+			req := motion.MoveOnMapReq{
+				ComponentName: base.Named("test-base"),
+				Destination:   easyGoalInSLAMFrame,
+				SlamName:      slam.Named("test_slam"),
+				Extra:         extra,
+			}
+
+			executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+
+			timeoutCtx, timeoutFn := context.WithTimeout(ctx, time.Second*5)
+			defer timeoutFn()
+			err = motion.PollHistoryUntilSuccessOrError(timeoutCtx, ms, time.Millisecond*5, motion.PlanHistoryReq{
+				ComponentName: req.ComponentName,
+				ExecutionID:   executionID,
+				LastPlanOnly:  true,
+			})
+			test.That(t, err, test.ShouldBeNil)
+
+			endPos, err := kb.CurrentPosition(ctx)
+			test.That(t, err, test.ShouldBeNil)
+
+			test.That(t, spatialmath.PoseAlmostEqualEps(endPos.Pose(), easyGoalInBaseFrame, 10), test.ShouldBeTrue)
+		})
+
+		t.Run("check that position-only mode executes", func(t *testing.T) {
+			// TODO(RSDK-5758): unskip this
+			t.Skip()
+			kb, ms := createMoveOnMapEnvironment(ctx, t, "pointcloud/octagonspace.pcd", 40, nil)
+			defer ms.Close(ctx)
+
+			req := motion.MoveOnMapReq{
+				ComponentName: base.Named("test-base"),
+				Destination:   goalInSLAMFrame,
+				SlamName:      slam.Named("test_slam"),
+				Extra:         extraPosOnly,
+			}
+
+			executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
```

Comment:
Could we please give this context a timeout so that the test doesn't hang if we introduce a deadlock into `MoveOnMapNew` later?